src/server/game/Entities/Creature/Creature.cpp

```diff
@@ -3343,8 +3343,20 @@ void Creature::AtEngage(Unit* target)
 
     MovementGeneratorType const movetype = GetMotionMaster()->GetCurrentMovementGeneratorType();
     if (movetype == WAYPOINT_MOTION_TYPE || movetype == POINT_MOTION_TYPE || (IsAIEnabled() && AI()->IsEscorted()))
+    {
         SetHomePosition(GetPosition());
 
+        // if its a vehicle, set the home positon of every creature passenger at engage
+        // so that they are in combat range if hostile
+        if (Vehicle* vehicle = GetVehicleKit())
+        {
+            for (auto seat = vehicle->Seats.begin(); seat != vehicle->Seats.end(); ++seat)
+                if (Unit* passenger = vehicle->GetPassenger(seat->first))
```

Comment:
if I check other usages of `Seats.begin()`, usually the Unit* is retrieved from `seat->second.Passenger.Guid`, like
```
if (Unit* passenger = ObjectAccessor::GetUnit(*this, seat->second.Passenger.Guid))
```
as vehicle->GetPassenger() does one extra lookup calling Seats.find()

Reply:
Done